src/reducers/pipelines.js

```diff
@@ -19,6 +19,14 @@ import { ALL_NAMESPACES } from '../constants';
 
 function byId(state = {}, action) {
   switch (action.type) {
+    case 'PipelineCreated':
+    case 'PipelineUpdated':
+      const runById = { [action.payload.metadata.uid]: action.payload };
```

Comment:
not a 'run', same below in byNamespace, in the tasks reducer, and some comments in the tests.

I know we'll be getting rid of a lot of this duplication soon (hopefully), but in the meantime we should make sure naming is correct to avoid any potential confusion.